featurematrix.go

```diff
@@ -442,6 +442,23 @@ func (fm *FeatureMatrix) LoadCases(data *csv.Reader, rowlabels bool) {
 
 }
 
+// Makes a copy of the feature matrix
+func (fm *FeatureMatrix) Copy() *FeatureMatrix {
+	newmap := make(map[string]int)
+	newdata := make([]Feature, len(fm.Data), len(fm.Data))
```

Comment:
[nit] typically you only set the capacity of a slice when you're initializing the length to zero.